src/libraries/MathLib.sol

```diff
@@ -47,11 +47,28 @@ library MathLib {
 
     /// @notice determines if input numbers have the same sign
     /// @dev asserts that both numbers are not zero
-    /// @param x signed number
-    /// @param y signed number
+    /// @param x signed 128-bit number
+    /// @param y signed 128-bit number
     /// @return true if same sign, false otherwise
     function isSameSign(int128 x, int128 y) internal pure returns (bool) {
         assert(x != 0 && y != 0);
         return (x ^ y) >= 0;
     }
+
+    /// @notice safely cast uint256 to int256
+    /// @dev reverts if the input is greater than or equal to 2^255
+    /// @param x unsigned 256-bit number
+    /// @return z signed 256-bit number
+    function toInt256(uint256 x) internal pure returns (int256) {
+        if (x >= 1 << 255) {
```

Comment:
**Thought:** Could also use `x > type(int256).max`, this is perhaps more readable. Not sure which is more gas efficient.